### DISCUSSION TITLE
No Servers Found There are no servers for the "stage-teleport" cluster.

### DISCUSSION THREAD

**Discussion by bravitejareddy** [gravitational/teleport#7418] [General]
Recently we have upgraded our teleport from 4.0.4 to 6.2.5, we modified our user add script to remove the **NOTE** below.

tctl users add "test-user" --roles=admin and restarted the teleport and after that when I log into the teleport UI I don't see node servers (both proxy and bastion). But when I issue the `tctl nodes ls` I see both the teleport-proxy and my bastion server.

[root@stage-teleport-proxy ~]# /usr/local/bin/tctl nodes ls
Nodename             UUID                                 Address           Labels
-------------------- ------------------------------------ ----------------- ---------------------------------------------------
STAGE-BASTION       xxxxxxx-xxxxxxx-xxxxxxx-xxxxx-xxxxxxxxxxxxxxx 192.20.1.20:3022  hostname=STAGE-BASTION,role=node,uptime=07:55:20 up 120 days, 17:29,  5 users,  load average: 0.16, 0.29, 0.34
stage-teleport-proxy 95e357e4-a3dc-4164-9b4d-24d30ef9fd18 192.20.0.57:3022 hostname=stage-teleport-proxy,role=node,uptime=07:51:40 up 9 days,  3:29,  2 users,  load average: 1.01, 0.96, 0.91

**NOTE**:
b'NOTE: Teleport 6.0 added RBAC in Open Source edition.\n\nIn the future, please create a role and use a new format with --roles flag:\n\n$ tctl users add "test-user" --roles=[add your role here]\n\nWe will deprecate the old format in the next release of Teleport.\nMeanwhile we are going to assign user "test-user" to role "admin" created during migration.\n\nUser "test-user" has been created but requires a password. Share this URL with the user to complete user setup, link is valid for 1h:\nhttps://stage-teleport.example:443/web/invite/xxxxxxxxxxxxxxxxxx\n\nNOTE: Make sure stage-teleport.example.com:443 points at a Teleport proxy which users can access.\n'

![teleport-no servers](https://user-images.githubusercontent.com/55269989/123589122-64f34500-d806-11eb-994a-9a2e12c89b0f.jpg)

**Comment by webvictim** [comment]
If you can't see certain nodes in the web UI but you can when running `tctl nodes ls` on the auth server, it's likely that your Teleport user doesn't have a role which grants it permission to view those nodes. Can you share the output of `tctl get user/<username>`, replacing `<username>` with the Teleport username you are using to log into the web UI?

**Reply by bravitejareddy** [reply]
This issue got fixed after we change the users creation in teleport from < tctl users add "test-user" --roles=admin > to our old way of creating teleport user < tctl users add test-user test-user >.

The only reason that we have changed the command to < tctl users add "test-user" --roles=admin > is to avoid the NOTE  which we are getting while creating the user with the old command < tctl users add test-user test-user > 

NOTE:

b'NOTE: Teleport 6.0 added RBAC in Open Source edition.\n\nIn the future, please create a role and use a new format with --roles flag:\n\n$ tctl users add " test-user" --roles=[add your role here]\n\nWe will deprecate the old format in the next release of Teleport.\nMeanwhile we are going to assign user " test-user" to role "admin" created during migration.\n\n'
